src/federated.c

```diff
@@ -280,21 +278,6 @@ void FederatedConnectionBundle_msg_received_cb(FederatedConnectionBundle *self,
   }
 }
 
-void FederatedConnectionBundle_network_channel_state_changed(FederatedConnectionBundle *self) {
-  NetworkChannelState state = self->net_channel->get_connection_state(self->net_channel);
-  switch (state) {
-  case NETWORK_CHANNEL_STATE_CONNECTED:
-  case NETWORK_CHANNEL_STATE_LOST_CONNECTION:
-    for (size_t i = 0; i < self->inputs_size; i++) {
-      FederatedInputConnection *input = self->inputs[i];
-      input->last_known_tag = FOREVER_TAG;
```

Comment:
I think I still don't understand the runtime fully enough, so I am not sure where to set this tag now instead

Reply:
We could set it in Scheduler_acquire_tag, there we can check the state of the input port before we go to sleep waiting for more messages on it. We can add that in the next PR